### PR TITLE
Add missing installation step for Windows

### PR DESCRIPTION
**Pull request by pieterdd**
Initially the instructions [didn't work for me](https://github.com/gtk-rs/gtk4-rs/discussions/1583), but I was able to figure out that I was missing a bit of linker configuration. I documented this requirement and rewrote other parts of the Windows documentation a little bit.

### CODE REVIEW

File: book/src/installation_windows.md [hunk 1]
```diff
@@ -16,26 +16,39 @@ Install the rust toolchain via [rustup](https://rustup.rs/).
...
+
+### Ensure your linker can find GTK's library files
+
+Open `%HOMEPATH%\.cargo\config` for editing. You can add extra flags for your linker here to tell it where GTK's library files are. If you're using LLVM's linker with the `stable-x86_64-pc-windows-msvc` Rust toolchain, your config may look like this:
+
+```toml
+[target.x86_64-pc-windows-msvc]
+linker = "lld-link.exe"
+rustflags = ["-C", "link-args=/LIBPATH:C:\\gtk-build\\gtk\\x64\\release\\lib\\"]
+```
```
Comment (Hofer-Julian) [inline book/src/installation_windows.md:51]: I assume that's only necessary if you use `lld`?
If yes, please make that clear here

**Review by Hofer-Julian — COMMENTED** [review]
One comment, apart from that, it looks good.
Thanks!

File: book/src/installation_windows.md (hunk 1 above)
Comment (pieterdd) [reply book/src/installation_windows.md:51]: You also need to use it when using the default linker, I just didn't research the appropriate flag to do so. Here's what happens when I remove my `~/.cargo/config`:

```
error: linking with `link.exe` failed: exit code: 1181
(...)
  = note: LINK : fatal error LNK1181: cannot open input file 'gobject-2.0.lib'
```

This works fine for me:

```toml
[target.x86_64-pc-windows-msvc]
rustflags = ["-C", "link-args=/LIBPATH:C:\\gtk-build\\gtk\\x64\\release\\lib\\"]
```

It looks like both linkers support the same flag, so I'll update the documentation accordingly.

Comment (Hofer-Julian) [reply book/src/installation_windows.md]: I wonder why I didn't have to do that, but oh well.
Let's add it :)

**Review by Hofer-Julian — APPROVED** [review]

**Pull request merged**